### ISSUE TITLE
В отчете постоянно пишет: Проверка не завершена [Дата время]

### ISSUE THREAD

**Issue by mszerg** [novostrim/watcher4site#11]
Во вкладве Отчеты пишет
Состояние: Проверка не завершена [2016-03-18 13:00:01]
Проверено файлов:    11996 [Изменения не обнаружены]
Сравнивать MD5-хэш:    Нет
Игнорировать расширения и файлы:    jpg

Почему не завтершена? и как посмотреть что не завершено?

Заранее спасибо.

**Comment by gentee** [comment]
Это вы это видите после запуска из крона? Дело в том, что есть ограничение по времени на выполнение PHP скриптов, как правило 30 секунд. Если файлов очень много, то из крона скрипт не успевает все проверить, поэтому и появляется такой статус. В следующий раз при запуске по крону он просто продолжает проверку с того места, где остановился. Ну или можно вручную продолжить проверку.

**Comment by mszerg** [comment]
Да, по крону, но судя по времени 13:00:01 Скрипт выполнялся всего 1секунду, запомнил время впредыдущих запусков, там тоже было 1-2 секунлы.Cron срабатывает каждый час.

Запустил скрипт вручную, сперва он выполнился за 2секунды, походу допроверил. Последующие разы где-то 11-12-секунд работал. После этого в отчете пишет:Проверка завершена
Посмотрю что будет через час, когда по крону сработает.

**Comment by gentee** [comment]
Может ограничения для крона еще какие-то есть.

**Comment by mszerg** [comment]
Поставил крон на выполнение кажлые 5минут.
В кроне прописана строка
php /home/,,,/public_html/watcher/api/cron.php >/home/,,,/public_html/watcher/watcher.log 2>&1

В итоге в лог он выдает
PHP Notice:  Undefined index: HTTP_HOST in /home/mszerg/web/autohome.org.ua/public_html/watcher/lib/lib.php on line 8
PHP Notice:  Undefined index: SERVER_ADDR in /home/mszerg/web/autohome.org.ua/public_html/watcher/lib/lib.php on line 9
0=0=0

Но он и до этого выдавал тоже самое, даже в командную строку. Но с командной строки отрабатывает нормально.

Выяснил, что полная проверка проходит каждый второй раз. Первый раз недопроверяет.
В php.ini время выполнения скриптов 30секунд. Проверяло примерно 11тыс файлов и останавливалось.
Увеличил время выполнения скриптов в php.ini сперва до 60сек потом до 120сек. Перезагрузил апач.
Стало проверять за раз примерно 12тыс файлов, но не добивает максимум 500.

Всего файлов 12368

Информации что крон имеет ограничение на время выполнения не нашел.

**Comment by mszerg** [comment]
Промониторил в htop момент запуска скрипта по крону 100% загрузка процесом mysql максимум 5 секунд

**Comment by gentee** [comment]
mysql и не должен грузить всё время на 100%. Там же еще сканирование диска идет.

**Comment by mszerg** [comment]
50% mysql и 50% скрипт cron.php, то я не успел расмотреть))
В общем я понаблюдал за зарузкой процессора и видно, что скрипт выполнятется ну максимум 15 секунд. Увеличение времени выполнения скриптов в php ничего не дает. Первое сканирование всегда колеблется от 11 до 12тыс файлов. Второе добивает оставшиеся 300-500 файлов.

**Comment by gentee** [comment]
Можете открыть базу MySQL в phpMyAdmin? Найдите там таблицу (префикс)_task, откройте и там должны увидеть вашу задачу. Увеличьте у нее значение поля limit - это количество секунд работы.

**Comment by mszerg** [comment]
Очень похоже, что это ключевой фактор. Потому как первый раз за 10сек скрипт не успевает, потов торой задачей за 2секугнды добивает остаток.
А как увеличить данное значение на постоянной основе, оно создает новую задачу и там всегда 10сек стоит.

**Comment by gentee** [comment]
Измените api/watcher.php - 230 строка
$pars['limit'] = 10;
Поставьте там нужное значение.

**Comment by mszerg** [comment]
Спасибо, поставил 20сек, работает.
Еще думаю вам нужно поставить в отчете время окончания выполнения скрипта, вместо времени его начала.
Состояние: Проверка завершена [2016-03-21 **13:00:02**]

Это более логично и информационно, тем более у вас в базе данных такое поле есть.

Вот мои значения из базы данных, таблица ws_task
start           2016-03-21 13:00:02
finish                2016-03-21 13:00:12

**Comment by gentee** [comment]
Я давно уже там ничего не правил. Как накопится несколько изменений, так и исправлю все разом.
